### PR TITLE
Update the style of the Modal component

### PR DESCRIPTION
**Pull request by thomasguillot**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

I noticed the `Modal` felt slightly off, especially with the header and the various paddings. This PR restyle it to use a 64px padding around it. It also moves the title more inline with the content. For buttons, we are now using a `buttonsCard`.

I had to reorganise things in the Campaigns to make sure the changes were not breaking the "Add New Prompt" modal. And as an added bonus, I've tweaked the "Add New Prompt" modal to something similar to what was suggested in #861 

### How to test the changes in this Pull Request:

1. Check the various Modals (Components Demo, Campaigns, Analytics -- Custom Events, etc...)
2. Switch to this branch
3. Test again

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: assets/components/src/modal/style.scss [hunk 1]
```diff
@@ -16,7 +16,7 @@
 	width: 100%;
```
Comment (dkoo) [inline assets/components/src/modal/style.scss:16]: Adding `max-height: 100%` ( or `100vh`) lets the modal content scroll if it exceeds the height of the viewport:

```suggestion
	width: 100%;
	max-height: 100%;
```

**Review by dkoo — CHANGES_REQUESTED** [review]
The new spacing looks good in Campaigns, components demo, and Analytics. However I noticed on **Analytics > Custom Events** the modal content exceeds the viewport size at smaller viewports. Tested the below and it seems to work.

Also, we'll want to publish this to the NPM package, right? I would love to figure out why the `@wordpress/icons` package [crashes](https://github.com/Automattic/newspack-popups/issues/556) when used in the NPM package before we publish the next version, though.

File: assets/components/src/modal/style.scss (hunk 1 above)
Comment (thomasguillot) [reply assets/components/src/modal/style.scss:17]: Fixed in https://github.com/Automattic/newspack-plugin/pull/1013/commits/a543fba520a2d7eaa83d76145f4de75a00e3dd5a

**Review by dkoo — CHANGES_REQUESTED** [review]
Oh, I'm a fan of this new "Add New Prompt" modal. It does occur to me that we have one additional placement, though—Manual Only. Is it a concern that this isn't one of the options in the modal?

Screen Shot 2021-07-02 at 3 28 20 PM

I suppose it's already missing from the existing modal and having 7 options doesn't lend itself well to a neat grid, plus we don't necessarily want to encourage users to create a lot of manual-only prompts, so I'm okay with omitting it. Marking "request changes" so we make sure to discuss, but if you don't think it's necessary then I'll approve.

**Comment by thomasguillot** [comment]
@dkoo I think we should add the "Manual Only" option. We should display all the available options. That being said, I'd suggest to do it in a separte PR since I will need to reorganise the Grid/Modal and I need to think about it 😊

**Review by dkoo — APPROVED** [review]

**Pull request merged**